src/cryptographic_primitives/polynomial.rs

```diff
@@ -0,0 +1,371 @@
+use std::convert::TryFrom;
```

Comment:
I don't think polynomial count as a cryptographic primitive, but I am not sure where else to put it

Reply:
I think it can be located in `crate::elliptic` module as well (as it heavily uses elliptic things). Currently elliptic module consist only of `curves` submodule. What do you think about moving it to elliptic module?

Reply:
my opinion is to move polynomial into cryptographic_primitives/secret_sharing

Reply:
Fair enough. Moved it into secret_sharing.